pulp_ansible/app/tasks/collections.py

```diff
@@ -46,92 +57,9 @@ def sync(remote_pk, repository_pk, mirror):
     if not remote.whitelist:
         raise ValueError(_("A CollectionRemote must have a 'whitelist' specified to synchronize."))
 
-    repository_spec_strings = remote.whitelist.split(" ")
-
-    def nowhere(*args, **kwargs):
-        pass
-
-    collection_version_pks = []
-    download_pb = ProgressBar(message="Downloading Collections", total=len(repository_spec_strings))
-    import_pb = ProgressBar(message="Importing Collections", total=len(repository_spec_strings))
-
-    with RepositoryVersion.create(repository) as new_version:
-        with tempfile.TemporaryDirectory() as temp_ansible_path:
-            with download_pb:
-                # workaround: mazer logs errors without this dir https://pulp.plan.io/issues/4999
-                os.mkdir(os.path.join(temp_ansible_path, "ansible_collections"))
-
-                galaxy_context = GalaxyContext(
-                    collections_path=temp_ansible_path,
-                    server={"url": remote.url, "ignore_certs": False},
-                )
-
-                install_repository_specs_loop(
-                    display_callback=nowhere,
-                    galaxy_context=galaxy_context,
-                    repository_spec_strings=repository_spec_strings,
-                )
-
-                download_pb.done = len(repository_spec_strings)
-
-            with import_pb:
-                content_walk_generator = os.walk(temp_ansible_path)
-                for dirpath, dirnames, filenames in content_walk_generator:
-                    if "MANIFEST.json" in filenames:
-                        manifest_path = os.path.join(dirpath, "MANIFEST.json")
-                        with open(manifest_path) as manifest_file:
-                            manifest_data = json.load(manifest_file)
-                        info = manifest_data["collection_info"]
-                        filename = "{namespace}-{name}-{version}".format(
-                            namespace=info["namespace"], name=info["name"], version=info["version"]
-                        )
-                        tarfile_path = os.path.join(temp_ansible_path, filename + ".tar.gz")
-                        with tarfile.open(name=tarfile_path, mode="w|gz") as newtar:
-                            newtar.add(dirpath, arcname=filename)
-
-                        with transaction.atomic():
-                            collection, created = Collection.objects.get_or_create(
-                                namespace=info["namespace"], name=info["name"]
-                            )
-                            collection_version, created = CollectionVersion.objects.get_or_create(
-                                collection=collection, version=info["version"]
-                            )
-
-                            if created:
-                                # Create the tags
-                                tags = info.pop("tags")
```

Comment:
where can I get this data?

Reply:
I expect that to come from the inspected manifest.json file:  https://github.com/ansible/mazer/blob/devel/tests/ansible_galaxy/collection_examples/hello/galaxy.yml

That data would show up here as the key 'tags' for example: https://github.com/pulp/pulp_ansible/blob/master/pulp_ansible/app/tasks/collections.py#L83

Reply:
how can I get manifest.json from the galaxy api?

Reply:
it's in the file that is downloaded. That downloading happens by the stages API. I think you want to inspect this file as aprt of the pre-save for content, or maybe the post-save.